### PR TITLE
[ARO] CLI enhancements, including route table checking permissions

### PR DESCRIPTION
**Pull request by kwoodson**
**Description**  

The updates here are the latest development features to be added to Azure Red Hat OpenShift az cli client.
- enhanced permissions validation
- logging improvements
- type checking

**Testing Guide**  

None
These are teh enhancements that we use for our Azure cli in the github.com/Azure/ARO-RP repository.

**History Notes**  

None
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by kwoodson** [comment]
@ehashman @jim-minter @mjudeikis 
I have updated the PR from #14362 after some recent updates.  Here is the latest.  PTAL

**Comment by kwoodson** [comment]
@petrkotas Looks like some errors on the roleassignments according to these [logs](https://dev.azure.com/azure-sdk/public/_build/results?buildId=473793&view=logs&j=74095127-2a27-5370-37ed-15a4193f243f&t=5cf5f89a-09fb-583c-72e4-4e4427c89fb1)

Also looks like the style failed because of the `AADManager(object)` inheritance.

**Comment by yonzhan** [comment]
ARO

**Comment by ehashman** [comment]
@kwoodson it looks like there are test conflicts, can you please pull the latest HEAD from az-cli and rebase/rerecord?

**Comment by kwoodson** [comment]
Rebased to old tests.  Will update tests as a second commit.

**Comment by kwoodson** [comment]
After rebasing and running the tests, I am running into some errors that I'm pretty sure aren't related to the PR.  Might be an issue with the dev branch?

```
src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py:113:
az network vnet create: error: APIVersion 2020-06-01 is not available
E           NotImplementedError: APIVersion 2020-06-01 is not available
```

**Comment by kwoodson** [comment]
```
$ azdev test aro --live
...
    def create_resource_group(cmd, rg_name, location, tags=None, managed_by=None):
        """ Create a new resource group.
        :param str resource_group_name:the desired resource group name
        :param str location:the resource group location
        :param str tags:tags in 'a=b c' format
        """
        rcf = _resource_client_factory(cmd.cli_ctx)
    
        ResourceGroup = cmd.get_models('ResourceGroup')
>       parameters = ResourceGroup(
            location=location,
            tags=tags
        )
E       TypeError: 'NoneType' object is not callable

src/azure-cli/azure/cli/command_modules/resource/custom.py:1074: TypeError
```

**Comment by kwoodson** [comment]
I have updated my azdev environment with instructions [here](https://github.com/Azure/azure-cli-dev-tools#setting-up-your-development-environment) and things seem to be back to normal.

Rerunning the tests.

**Comment by kwoodson** [comment]
Looks like we need to purge some cruft before rerunning these tests.
```
E           azure.graphrbac.models.graph_error_py3.GraphErrorException: The directory object quota limit for the Principal has been exceeded. Please ask your administrator to increase the quota limit or delete objects to reduce the used quota.
```

**Comment by petrkotas** [comment]
@kwoodson I went thought he PR and did not find anything related to issues I have encoutered. Judging by you last comment, the PR is fine now? If not let me know, I will continue the search.

**Comment by kwoodson** [comment]
@petrkotas Hitting this error message.  Seems I've exhausted the testing infrastructure and require someone to remove the objects.  

```
E           azure.graphrbac.models.graph_error_py3.GraphErrorException: The directory object quota limit for the Principal has been exceeded. Please ask your administrator to increase the quota limit or delete objects to reduce the used quota.
```

**Comment by kwoodson** [comment]
Cleaning up personal application registrations seems to have unblocked.  Running tests.

**Comment by kwoodson** [comment]
After performing some debug I determined that the local tests were using my custom extension instead of the azure-cli extension code.  This was determined while walking pdb.  This was making it so the mock was not mocking the _gen_uuid call.  Once I recognized this, I removed the path to my extension and was able to successfully run the tests and replay them successfully.

```
======= 9 passed, 1 warning in 5.15s =====
```

**Comment by kwoodson** [comment]
Failing on this error:
```
ModuleNotFoundError: No module named 'azure.mgmt.redhatopenshift.operations'
```

I attempted to replace this with `azure.mgmt.redhatopenshift.v2020_04_30.operations' but the error returned with `azure.mgmt.redhatopenshift.v2020_04_30.v2020_04_30` which was duplicated.  I belive this is due to this code here:
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/redhatopenshift/azure-mgmt-redhatopenshift/azure/mgmt/redhatopenshift/_azure_red_hat_open_shift_client.py#L103

I'll continue poking at this.

@petrkotas, lmk if you have any ideas.  Thanks

**Comment by kwoodson** [comment]
rebased back on top of dev branch.  let's try this CI stuff again!

**Comment by kwoodson** [comment]
@qwordy Would you mind assisting with this? 

I created an issue https://github.com/Azure/azure-cli/issues/14614.  I'm not sure what I'm missing or if there is something in the CI that requires attention.

Thanks!

File: src/azure-cli/azure/cli/command_modules/aro/__init__.py [hunk 1]
```diff
@@ -20,8 +19,7 @@ def __init__(self, cli_ctx=None):
             operations_tmpl='azure.cli.command_modules.aro.custom#{}',
             client_factory=cf_aro)
         super(AroCommandsLoader, self).__init__(cli_ctx=cli_ctx,
-                                                custom_command_type=aro_custom,
-                                                resource_type=ResourceType.MGMT_ARO)
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/aro/__init__.py:24]: Why removing this line?

Comment (kwoodson) [reply src/azure-cli/azure/cli/command_modules/aro/__init__.py]: I added this back.

**Comment by kwoodson** [comment]
@qwordy @haroldrandom Thanks for the suggestion.  I think this is ready to go.

**Comment by ehashman** [comment]
@mjudeikis can you PTAL? I didn't get a chance today

**Comment by kwoodson** [comment]
@fengzhou-msft  @qwordy @haroldrandom @mjudeikis PTAL

**Comment by mjudeikis** [comment]
lgtm

**Comment by julienstroheker** [comment]
@yonzhan can we get this merged ? thanks

**Comment by yonzhan** [comment]
@qwordy please help with this PR.

**Pull request merged**